### ISSUE TITLE
Migrate URLs as Former URLs

### ISSUE THREAD

**Issue by bluemodus-sandro** [Kentico/xperience-by-kentico-kentico-migration-tool#310]
### Motivation

I am migrating navigable pages from Xperience 13 into XbyK and would like to preserve the original URLs. However, the URLs were created using a fairly complex custom macro in the source project. After the migration, the tree fails to load in the target unless I switch back to content tree routing and rerun the migration. 

### Proposed solution

Similar to the feature requested here: https://github.com/Kentico/xperience-by-kentico-universal-migration-tool/issues/77

I would like to be able to use the ClassMapping classes to add the URLs from the source as former URLs. Additionally, when that feature is enabled or in use for a specific type, I would like for the migration tool to generate new URLs, or transform the old ones into something compatible\functional in the XbyK project.

**Comment by Sapphirress** [comment]
Hi @twozero6 , we pushed back enhancements in our priorities due to more pressing bugs and other issues. But as the bugs have been mostly solved, we are back in the game for adding improvements. We will keep you posted. Thanks for keeping us busy 👍

**Comment by akfakmot** [comment]
@bluemodus-sandro Hi, 

do you think the following extension of ContentItemDirector solves your problem? If not, can you specify what else you need?

2 new methods in ContentItemDirector:
    - ```RegenerateUrlPath```, which lets the system generate default paths (in all cultures) if called for the directed item.
    - ```OverrideFormerUrlPaths```, which allows you to do what the name says. Source former URL paths are passed by 
the ContentItemSource parameter, so it's possible to do whatever you want - clear them, add new ones and keep old ones, etc.

I imagine you could add the original path using the ```OverrideFormerUrlPaths``` (concat on the default ones ContentItemSource)  and then call ```RegenerateUrlPath``` to assure a normal default URL. 

However I'm not exactly sure how the original "complex" URL is stored in the source instance. Can you give me more details here?
Basically you could mimic what ```MigratePageUrlPaths``` method does for your concrete scenario. It's possible to inject the auxiliary objects used by ```MigratePageUrlPaths``` even into your director (sourceInstanceContext, modelFacade, ...)

**Comment by bluemodus-sandro** [comment]
Hi @akfakmot,

What you described would solve our problem. If we can access the source context, we can bring over the URLs or reconstruct them the way we need them to be. By "complex" URLs, I meant that they are generated by a custom macro that has complicated logic - the URLs get generated based on things like assigned categories instead of their location in the tree or the nodealiaspath.

**Comment by akfakmot** [comment]
@bluemodus-sandro  Thank you for the confirmation, I'll post a PR you can try in an hour or two.

**Comment by akfakmot** [comment]
@bluemodus-sandro https://github.com/Kentico/xperience-by-kentico-kentico-migration-tool/pull/371

We'll include the feature in official release next week.
